src/platform/BUILD.gn

```diff
@@ -30,7 +30,7 @@ if (chip_enable_openthread) {
   }
 }
 
-if (chip_device_platform == "linux" && chip_mdns != "none") {
+if (chip_device_platform == "linux" && chip_mdns == "platform") {
   pkg_config("avahi_client_config") {
```

Comment:
Ideally this would be removed, since it is a duplicate of the one in src/patform/Linux/BUILD.gn